pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstorecache_test.go

```diff
@@ -0,0 +1,444 @@
+				Content:        "content1",
+				LogID:          "1111111111111111",
+				IntegratedTime: "1111111111111111",
+			},
+		},
+	}
+
+	selectors2 = Item{
+		Key: "signature2",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex2@example.com",
+				Content:        "content2",
+				LogID:          "2222222222222222",
+				IntegratedTime: "2222222222222222",
+			},
+		},
+	}
+
+	selectors3 = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content3",
+				LogID:          "3333333333333333",
+				IntegratedTime: "3333333333333333",
+			},
+		},
+	}
+
+	selectors3Updated = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content4",
+				LogID:          "4444444444444444",
+				IntegratedTime: "4444444444444444",
+			},
+		},
+	}
+
+	selectors2Updated = Item{
+		Key: "signature2",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex2@example.com",
+				Content:        "content5",
+				LogID:          "5555555555555555",
+				IntegratedTime: "5555555555555555",
+			},
+		},
+	}
+)
+
+func TestNewCache(t *testing.T) {
+	want := &cacheImpl{
+		size:     3,
+		items:    list.New(),
+		mutex:    sync.RWMutex{},
+		itemsMap: make(map[string]MapItem),
+	}
+	got := NewCache(3)
+	require.Equal(t, want, got, "NewCache() = %v, want %v", got, want)
+}
+
+func TestCacheimpl_GetSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	m[selectors1.Key] = MapItem{
+		item:    &selectors1,
+		element: items.PushFront(selectors1.Key),
+	}
+	m[selectors2.Key] = MapItem{
+		item:    &selectors2,
+		element: items.PushFront(selectors2.Key),
+	}
+
+	cacheInstance := &cacheImpl{
+		size:     3,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name         string
+		want         *Item
+		key          string
+		errorMessage string
+	}{
+		{
+			name:         "Non existing entry",
+			want:         nil,
+			key:          selectors3.Key,
+			errorMessage: "A non-existing item's key should return a nil item.",
+		},
+		{
+			name:         "Existing entry",
+			want:         &selectors1,
+			key:          selectors1.Key,
+			errorMessage: "An existing items key's should return the existing item",
+		},
+	}
+
+	for _, tt := range tests {
+		got := cacheInstance.GetSignature(tt.key)
+		require.Equal(t, got, tt.want, "%v Got: %v Want: %v", tt.errorMessage, got, tt.want)
+	}
+}
+
+func TestCacheimpl_PutSignature(t *testing.T) {
+	m := make(map[string]MapItem)
+	items := list.New()
+
+	cacheInstance := &cacheImpl{
+		size:     2,
+		items:    items,
+		mutex:    sync.RWMutex{},
+		itemsMap: m,
+	}
+
+	tests := []struct {
+		name       string
+		item       *Item
+		wantLength int
+		wantKey    string
+		wantValue  *Item
+	}{
+		{
+			name:       "Put first element",
```

Comment:
this progressive approach can result on unexpected results on test cases, each test case must not affect the next one.

So I think you will need to create a new cache on each test case, and provide that cache with the values you need to have before it runs